wallettool/build.gradle

```diff
@@ -55,6 +55,11 @@ if (GradleVersion.current() > GradleVersion.version("8.0")) {
     application {
         mainClass = walletToolMainClassName
         applicationName = walletToolName
+        // isModularBuild is not strictly necessary here because we're already Gradle 8+
+        if (isModularBuild) {
```

Comment:
If it's not necessary, then why use it? I'm not against it, but I'd like to know why.

Reply:
When applications run on the module path they have stricter integrity guarantees.  E.g. if a module is accessing `sun.misc.unsafe` (in module `jdk.unsupported`) this will result in a warning or an error. It was by running wallet-tool as a module that I discovered that ProtoBuf is doing this. So enforcing these checks will help us make our modules better as well as warn us about bad behavior in 3rd-party modules.

Reply:
Looking at it with fresh eyes, it is necessary. Because if `isModularBuild` is set to false (for some reason) then setting `mainModule` is wrong.

My comment saying it was "unnecessary" actually meant that in our _current_ build configurations `isModularBuild` will never be false at this point in the code.

Reply:
Good catch! Updated and force-pushed.